src/app.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useReducer } from 'react';
+import { st, classes } from './app.st.css';
+import { TodoList } from './components/TodoList';
+import { AddTodo } from './components/AddTodo';
+import type { Todo } from './models/todo.model';
+import { Header } from './header';
+
+export interface AppProps {
+    className?: string;
+}
+
+export const ACTIONS = {
+    ADD_TODO: 'addTodo',
+    DELETE_TODO: 'deleteTodo',
+};
+
+export type Actions =
+    | { type: 'addTodo'; payload: { id: string; text: string } }
+    | { type: 'deleteTodo'; payload: { id: string } };
+
+export interface Action {
+    type: string;
+    payload?: Todo;
+}
+
+function reducer(todos: Todo[], action: Actions): Todo[] {
+    switch (action.type) {
+        case 'addTodo':
+            return [...todos, action.payload];
+        case 'deleteTodo':
+            return todos.filter((todo) => todo.id !== action.payload.id);
```

Comment:
+ another comment - 
maybe this `filter` logic should sit in the `models` level? 
the ID IMO should be private property